src/Components/Inputs/Select.php

```diff
@@ -90,7 +99,7 @@ protected function processValidations(string $fieldKey, string $fieldLabel, mixe
         $rules->addIfNotNull('required', $this->validation('required'));
         $rules->addIf('nullable', ! $this->validation('required'));
 
-        $rules->addIf(Rule::in($this->optionValues()), $this->dataSource === self::DATA_SRC_VALUES);
+        $rules->addIf(Rule::in($this->optionValues()), $this->dataSource === self::DATA_SRC_VALUES || $this->dataSource === self::DATA_SRC_RESOURCE);
```

Comment:
This will only work if the resource has a static list it can look up. If the resource is in front of an API (like directory search), it cannot contain all the possible valid values.